NPE in FileDataSecretEnricher.createSecretFromAnnotations

<!-- Please fill out the following information to help us in analyzing the issue, but feel free to skip it if you don't have the information at hand or if it does not apply. Please remove every section which does not apply to the issue. -->
### Description
In using the recent 4.4.0 release of fmp, I get an NPE in a project.  Here is the generated stacktrace:

```
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal io.fabric8:fabric8-maven-plugin:4.4.0:resource failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.NullPointerException
    at io.fabric8.maven.enricher.standard.FileDataSecretEnricher.createSecretFromAnnotations (FileDataSecretEnricher.java:63)
    at io.fabric8.maven.enricher.standard.FileDataSecretEnricher.access$000 (FileDataSecretEnricher.java:33)
    at io.fabric8.maven.enricher.standard.FileDataSecretEnricher$1.visit (FileDataSecretEnricher.java:53)
    at io.fabric8.maven.enricher.standard.FileDataSecretEnricher$1.visit (FileDataSecretEnricher.java:47)
    at io.fabric8.kubernetes.api.builder.BaseFluent.acceptInternal (BaseFluent.java:118)
    at io.fabric8.kubernetes.api.builder.BaseFluent.accept (BaseFluent.java:109)
    at io.fabric8.kubernetes.api.builder.BaseFluent.accept (BaseFluent.java:11)
    at io.fabric8.kubernetes.api.builder.BaseFluent.acceptInternal (BaseFluent.java:115)
    at io.fabric8.kubernetes.api.builder.BaseFluent.accept (BaseFluent.java:109)
    at io.fabric8.maven.enricher.standard.FileDataSecretEnricher.addAnnotations (FileDataSecretEnricher.java:47)
    at io.fabric8.maven.enricher.standard.FileDataSecretEnricher.create (FileDataSecretEnricher.java:43)
    at io.fabric8.maven.plugin.enricher.EnricherManager.lambda$createDefaultResources$1 (EnricherManager.java:72)
    at io.fabric8.maven.plugin.enricher.EnricherManager.loop (EnricherManager.java:104)
    at io.fabric8.maven.plugin.enricher.EnricherManager.createDefaultResources (EnricherManager.java:71)
    at io.fabric8.maven.plugin.enricher.EnricherManager.createDefaultResources (EnricherManager.java:66)
    at io.fabric8.maven.plugin.mojo.build.ResourceMojo.generateAppResources (ResourceMojo.java:493)
    at io.fabric8.maven.plugin.mojo.build.ResourceMojo.generateResources (ResourceMojo.java:452)
    at io.fabric8.maven.plugin.mojo.build.ResourceMojo.executeInternal (ResourceMojo.java:360)
    at io.fabric8.maven.plugin.mojo.AbstractFabric8Mojo.execute (AbstractFabric8Mojo.java:67)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

The exact same project works fine with 4.3.1.

### Info

* f-m-p version : 4.4.0
* Maven version (`mvn -v`) :
```
Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T09:06:16-06:00)
Maven home: /usr/local/Cellar/maven/3.6.2/libexec
Java version: 11.0.1, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/jdk-11.0.1+13/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.14.6", arch: "x86_64", family: "mac"
```
* Kubernetes / OpenShift setup and version :  I've been using minishift with OpenShift 3.11, but you do not need a cluster running to replicate.  Just need to execute the fabric8:resource goal to replicate the problem.

* If it's a bug, how to reproduce :
Using the project below, you can replicate the problem by editing the pom.xml file and change the version of fmp from 4.3.1 to 4.4.0.  Save the changes and execute: mvn clean fabric8:resource -Popenshift which will fail.  You can add the -e flag to see the stack trace I included in this bug report.

* If it's a feature request, what is your use case :

* Sample project : https://github.com/rdruss/vertx-cache-example.git

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,6 +25,7 @@ Usage:
 ### 4.5-SNAPSHOT
 * Fix #1789: script to extract changelog information for notifications
 * Fix #1770: Fixed my mistake, remove constructors from OpenshiftBuildConfig which were causing PluginConfigurationException
+* Fix #1791: Fix NullPointerException in FileDataSecretEnricher (due to recent change in Kubernetes Client Builder null to LinkedHashMap conversion)
 
 ### 4.4.0 (2020-02-13)
 * Fix #1572: Support maven --batch-mode option

--- a/enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/FileDataSecretEnricher.java
+++ b/enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/FileDataSecretEnricher.java
@@ -50,8 +50,10 @@ public class FileDataSecretEnricher extends BaseEnricher {
             public void visit(SecretBuilder element) {
                 final Map<String, String> annotations = element.buildMetadata().getAnnotations();
                 try {
-                    final Map<String, String> secretAnnotations = createSecretFromAnnotations(annotations);
-                    element.addToData(secretAnnotations);
+                    if (annotations != null && !annotations.isEmpty()) {
+                        final Map<String, String> secretAnnotations = createSecretFromAnnotations(annotations);
+                        element.addToData(secretAnnotations);
+                    }
                 } catch (IOException e) {
                     throw new IllegalArgumentException(e);
                 }

--- a/enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/SecretEnricher.java
+++ b/enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/SecretEnricher.java
@@ -56,17 +56,19 @@ public abstract class SecretEnricher extends BaseEnricher {
             @Override
             public void visit(SecretBuilder secretBuilder) {
                 Map<String, String> annotation = secretBuilder.buildMetadata().getAnnotations();
-                if (!annotation.containsKey(getAnnotationKey())) {
-                    return;
+                if (annotation != null) {
+                    if (!annotation.containsKey(getAnnotationKey())) {
+                        return;
+                    }
+                    String dockerId = annotation.get(getAnnotationKey());
+                    Map<String, String> data = generateData(dockerId);
+                    if (data == null) {
+                        return;
+                    }
+                    // remove the annotation key
+                    annotation.remove(getAnnotationKey());
+                    secretBuilder.addToData(data);
                 }
-                String dockerId = annotation.get(getAnnotationKey());
-                Map<String, String> data = generateData(dockerId);
-                if (data == null) {
-                    return;
-                }
-                // remove the annotation key
-                annotation.remove(getAnnotationKey());
-                secretBuilder.addToData(data);
             }
         });
 

